File: backend-dotnet/EventModelingGitHubCloneDotNet/EventModelingGitHubCloneDotNet/Shared/Domain/RepositoryId.cs
namespace EventModelingGitHubCloneDotNet.Shared.Domain
{
    public class RepositoryId
    {
        public string Owner { get; }
        public string Repository { get; }

        public RepositoryId(string owner, string repository)
        {
            Owner = owner;
            Repository = repository;
        }

        public static RepositoryId FromString(string raw)
        {
            var ownerAndRepository = raw.Split("+");
            return new RepositoryId(ownerAndRepository[0], ownerAndRepository[1]);
        }

        public override string ToString()
        {
            return $"{Owner}+{Repository}";
        }
    }
}

File: backend-kotlin/src/main/kotlin/pl/zycienakodach/eventmodeling/githubclone/shared/domain/RepositoryId.kt
package pl.zycienakodach.eventmodeling.githubclone.shared.domain

data class RepositoryId(val owner: String, val repository: String) {
    override fun toString(): String = "${owner}+$repository"

    companion object {
        fun fromString(raw: String): RepositoryId {
            val ownerAndRepository = raw.split('+')
            return RepositoryId(ownerAndRepository[0], ownerAndRepository[1]);
        }
    }
}
